### ISSUE TITLE
Add loom tests

### ISSUE THREAD

**Issue by arthurprs** [austinjones/postage-rs#37]
Loom https://github.com/tokio-rs/loom

This can be a significant amount of work but in my experience it is MUCH more effective catching concurrency bugs compared to stress tests.

Eg. https://github.com/tokio-rs/tokio/blob/master/tokio/src/sync/tests/loom_broadcast.rs

**Comment by austinjones** [comment]
This is a great idea!  Thanks for the suggestion 💯

**Comment by austinjones** [comment]
Hmm, there is an obstacle here.  I use the [atomic](https://crates.io/crates/atomic) crate a lot for atomic enums.  Unfortunately, it doesn't support loom.  I think I'll need to write my own enum wrapper in order to implement this.

**Comment by arthurprs** [comment]
Yeah, it's a very invasive test suite. There's also the crossbeam-queue data structures.

Tokio uses a trick that imports all primitives from a loom module, then swaps its contents based on a cfg flag https://github.com/tokio-rs/tokio/blob/master/tokio/src/loom/mod.rs but that breaks as soon as you have a single external crate :weary:
